zebedee-reader/src/main/java/com/github/onsdigital/zebedee/search/indexing/content/ZebedeeContentIndexer.java

```diff
@@ -0,0 +1,203 @@
+                                    .addParameter("line", line)
+                                    .log();
+                            return null;
+                        }
+                        String[] terms = split[3].split(" *, *");
+                        if (terms.length == 0) {
+                            logInfo("Skipping invalid external department")
+                                    .addParameter("line", line)
+                                    .log();
+                            return null;
+                        }
+
+                        return new Department(split[0], split[1], split[2], terms);
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
+
+            return departments;
+        }
+    }
+
+    /**
+     * Generates a new index name with time stamp
+     * @return
+     */
+    protected String getOnsIndexName() {
+        return SearchConfiguration.getSearchAlias() + System.currentTimeMillis();
+    }
+
+    private static List<String> resolveSearchTerms(String uri) {
+        if (null == uri) {
+            return null;
+        }
+
+        logDebug("Resolving terms for uri")
+                .addParameter("uri", uri)
+                .log();
+
+        SearchBoostTermsResolver resolver = SearchBoostTermsResolver.getSearchTermResolver();
+        List<String> terms = resolver.getTerms(uri);
+
+        String[] segments = uri.split("/");
+        for (String segment : segments) {
+            String documentUri = "/" + segment;
+            List<String> prefixTerms = resolver.getTermsForPrefix(documentUri);
+            if (null != prefixTerms) {
+                terms.addAll(prefixTerms);
+            }
+        }
+        return terms;
+    }
+
+    /**
+     * Loads the index-config.yml file from disk
+     * @return
+     */
+    public static Settings getSettings() {
+        Settings.Builder settingsBuilder = Settings.builder().
+                loadFromStream("index-config.yml", Indexer.class.getResourceAsStream("/search/index-config.yml"));
+        elasticSearchLog("Index settings").addParameter("settings", settingsBuilder.internalMap());
+        return settingsBuilder.build();
+    }
+
+    /**
+     * Loads settings for departments index from departments-index-config.yml file
+     * @return
+     */
+    public static Settings getDepartmentsSettings() {
+        Settings.Builder settingsBuilder = Settings.builder().
+                loadFromStream("departments-index-config.yml", Indexer.class.getResourceAsStream("/search/departments/departments-index-config.yml"));
+        return settingsBuilder.build();
+    }
+
+    /**
+     * Load the default ONS index mapping
+     * @return
+     */
+    public static String getDefaultMapping() throws IOException {
+        InputStream mappingSourceStream = Indexer.class.getResourceAsStream("/search/default-mapping.json");
```

Comment:
Should this stream be in a `try-with-resources` block? I don't think we want to leave this open.